plsync/planetlab/sync.py

```diff
@@ -872,3 +873,99 @@ def GetBootimage(hostname, imagetype="iso"):
     f = open(fname, 'w')
     f.write(bindata)
     f.close()
+
+
+def SyncSlice(sslice, hostname_or_site, addwhitelist, addsliceips, addusers,
+              createslice):
+    """Creates and/or Updates a slice object in the PLC DB."""
+    if createslice:
+        print "Making slice! %s" % sslice['name']
+        MakeSlice(sslice['name'])
+    SyncSliceExpiration(sslice['name'])
+
+    if addusers:
+        SyncPersonsOnSlice(sslice['name'], sslice['users'])
+
+    for attr in sslice['attrs']:
+        SyncSliceAttribute(sslice['name'], attr)
+    for h,node in sslice['network_list']:
+        if (hostname_or_site is None or
+            hostname_or_site == h or
+            hostname_or_site in h ):
```

Comment:
Done.

Yeah, I did some light cleanup during the switch. The entire file needs a little more attention.